### PR TITLE
Add website status page & improve error messages

### LINKED ISSUE #205: Website Status
Create website status page (`/website_status`).

### LINKED ISSUE #2095: Improve Error Reporting
At the moment, it can be unclear why the website has failed to process a request / doesn't work
e.g. Is the user disconnected from internet? Is the server down for maintenance, etc...

Therefore, we should improve the toast error messages beyond "Connection Failure".

We should use the [window.onLine](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine) api to test if a user has internet connection and display a more informative error message such as "Connection Failure. Please check your internet connection." if the user doesn't have internet.

We should also make a query to the api's `status` endpoint to check if the server is down for maintenance.

**Additional Information**

This has been a pain point for users as the client is semi-usable without an internet connection (after initial page load). Causing them to think that the server is down.

Related to: #205

### PR DESCRIPTION
**Pull request by hudson-newey**
# Add website status page & improve error messages

At the moment, it can be unclear why a request failed and the most common user error is a faulty VPN connection / no internet connection.

This pull request hopes to address these issues by improving "timed out" error messages and adding a "Website Status" page (`/website_status`) so that users can easily see if the problem is on their side or server side.

## Changes

* Add "website status" page
* Add models & services for `WebsiteStatus` models & endpoints
* Improve "timed out" error messages to tell the user if they are not connected to the internet
* Update maintainer on Docker image
* Remove deprecated libssl1.1 from Docker image
* Some highlight colours were hard coded and did not use the website theme (eg. links at [https://www.ecosounds.org/about/contact_us](https://www.ecosounds.org/about/contact_us) use the development colours because they are hard coded. These have been changed to use the theme colours
* Add tests for all the above

## Problems

None

## Issues

Fixes: #205 
Fixes: #2095 

## Visual Changes

image
**New "website status" page**

image
**New toast that displays if the users doesn't have internet connection**

## Final Checklist

- [x] Assign reviewers if you have permission
- [x] Assign labels if you have permission
- [x] Link issues related to PR
- [x] Ensure project linter is not producing any warnings (`npm run lint`)
- [x] Ensure build is passing on all browsers (`npm run test:all`)
- [x] Ensure CI build is passing
- [ ] Ensure docker container is passing ([docs](https://github.com/QutEcoacoustics/workbench-client#docker))

### CODE REVIEW

File: src/app/components/website-status/website-status.component.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  private healthyUnhealthyText(healthy?: boolean): string {
+    // if we don't receive a response from the sever (the sever is down) or the user doesn't have internet
+    // the "healthy" field will be "null". In this case, we want to display "Unknown" in the template
+    if (!isInstantiated<boolean>(healthy)) {
+      return "Unknown";
+    }
+
+    return healthy ? "Healthy" : "Unhealthy";
```
Comment (atruskie) [inline src/app/components/website-status/website-status.component.ts]: semantic colors of the badges might look ok?

File: src/app/services/baw-api/api.interceptor.service.ts [hunk 2]
```diff
@@ -127,12 +127,23 @@ export class BawApiInterceptor implements HttpInterceptor {
...
     // Response timed out
     if (response.status === CLIENT_TIMEOUT) {
-      const error = new BawApiError(
-        CLIENT_TIMEOUT,
+      // check if this is the fault of the server or the client
+      const clientHasInternet = navigator.onLine;
+
+      const noClientInternetMessage =
+        "You are not be connected to the internet. " +
+        "Please check your internet / VPN connection and try again.";
```
Comment (atruskie) [inline src/app/services/baw-api/api.interceptor.service.ts]:
```suggestion
        "Please check your internet (or VPN) connection and try again.";
```

File: src/app/services/baw-api/website-status/website-status.service.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    @Inject(API_ROOT) private apiRoot: string
+  ) {}
+
+  // if the server is fully down, we will not receive a response from the API
+  // we can therefore not determine the status of the server and we should return "null"
+  public show(): Observable<WebsiteStatus | null> {
+    const endpoint = `${this.apiRoot}/status`;
+
+    return this.http
+      .get<WebsiteStatus>(endpoint)
```
Comment (atruskie) [inline src/app/services/baw-api/website-status/website-status.service.ts]: does this send the `accept: application/json` header?

File: src/app/services/baw-api/website-status/website-status.service.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      .pipe(
+        map((response) => new WebsiteStatus(response)),
+        catchError(() => of(null))
```
Comment (atruskie) [inline src/app/services/baw-api/website-status/website-status.service.ts]: does the error have any information we're discarding here?

File: src/app/components/statistics/pages/statistics.component.spec.ts
Comment (atruskie) [inline src/app/components/statistics/pages/statistics.component.spec.ts:1]: redundant changes should be discarded

**Review by atruskie — CHANGES_REQUESTED** [review]
Looking good.

Maybe let's add a status link in the footer?

File: src/app/services/baw-api/website-status/website-status.service.ts (hunk 3 above)
Comment (hudson-newey) [reply src/app/services/baw-api/website-status/website-status.service.ts]: No it didn't

Fixed

image

File: src/app/services/baw-api/website-status/website-status.service.ts (hunk 4 above)
Comment (hudson-newey) [reply src/app/services/baw-api/website-status/website-status.service.ts]: We discard the [`BawApiError`](https://github.com/QutEcoacoustics/workbench-client/blob/13ad6e6347ee18b8160f95de046951fe4a265368/src/app/helpers/custom-errors/baw-api-error.ts#L7) information.

image

I'll make this service return errors and move error handling to the `website-status.component.ts`

File: src/app/components/website-status/website-status.component.ts (hunk 1 above)
Comment (hudson-newey) [reply src/app/components/website-status/website-status.component.ts]: Done

image

_Updated screenshot with semantic colours_

**Review by atruskie — APPROVED** [review]

File: src/app/services/baw-api/website-status/website-status.service.ts (hunk 4 above)
Comment (atruskie) [reply src/app/services/baw-api/website-status/website-status.service.ts]: you can still return null; but the golden rule is don't squash errors. at a minimum log an error, better yet handle it somehow...

**Pull request merged**